api/api.go

```diff
@@ -0,0 +1,48 @@
+package api
+
+import (
+	"fmt"
+	"net/http"
+
+	"github.com/SkynetLabs/malware-scanner/clamav"
+	"github.com/SkynetLabs/malware-scanner/database"
+	"github.com/julienschmidt/httprouter"
+	"github.com/sirupsen/logrus"
+	"gitlab.com/NebulousLabs/errors"
+)
+
+// API is our central entry point to all subsystems relevant to serving requests.
+type API struct {
+	staticDB     *database.DB
+	staticClamav *clamav.ClamAV
+	staticRouter *httprouter.Router
+	staticLogger *logrus.Logger
+}
+
+// New creates a new API instance.
+func New(db *database.DB, clam *clamav.ClamAV, logger *logrus.Logger) (*API, error) {
+	if db == nil {
+		return nil, errors.New("no DB provided")
+	}
+	if logger == nil {
+		logger = logrus.New()
```

Comment:
I think you should error here. If you need it for testing, just create a helper `newTestAPI` that creates the logger for the tests.

Reply:
Done.